content/livesync/models.textile

```diff
@@ -0,0 +1,356 @@
+
+A @model@ is a data model representation of a specific part of the frontend application. Each frontend client can have multiple data models within the same application.
+
+When creating a new Model using the Models SDK you provide two functions to the Model; a @sync()@ function and a @merge()@ function. The @sync()@ function is used by the SDK to retrieve the current state of the data model from your backend, and the @merge()@ function is used by the SDK to merge state change events published by the Database Connector with the existing frontend state in the Model. Both of these functions will be covered more later in this page.
+
+The following diagram provides a simplified overview of the Models SDK:
+
+TODO: Diagram of flow between backend, models SDK and the user interface
+
+h2(#authenticate). Authenticate
+
+An "API key":/auth#api-keys is required for your frontend app to authenticate with Ably. API keys are used either to authenticate directly with Ably using "basic authentication":/auth/basic, or to generate tokens for untrusted clients using "token authentication":/auth/token.
+
+<aside data-type='important'>
+  <p>The code examples shown as part of the documentation use "basic authentication":/auth/basic to demonstrate features for convenience. In your own applications, basic authentication should never be used on the client-side as it exposes your Ably API key. Instead use "token authentication":/auth/token. </p>
+</aside>
+
+"Sign up":https://ably.com/sign-up to Ably to create an API key in the "dashboard":https://ably.com/dashboard or use the "Control API":/account/control-api to create an API key programmatically.
+
+API keys and tokens have a set of "capabilities":/auth/capabilities assigned to them that specify which operations, such as @subscribe@ or @publish@ can be performed on which resources. To use the Models SDK, the API key requires the following capabilities, these two capabilities are discussed further down the page, but for the time being set these two and you can review them further along:
+
+* @subscribe@ for the channels you intend to subscribe to.
+* @history@ if you intend to "sync from historical messages":#replay-from-history.
+
+h2(#install). Install
+
+The Models SDK requires a realtime client created using the "Ably JavaScript SDK":/getting-started/setup to interact with the Ably service.
+
+Install the Ably JavaScript SDK and the Models SDK from "NPM:":https://www.npmjs.com
+
+```[sh]
+npm install ably @ably-labs/models
```

Comment:
The user will have to install the ably package as well if working in a project that doesn't have ably already.

Reply:
The ably package is included in the install.